rs/execution_environment/src/execution_environment.rs

```diff
@@ -2906,6 +2906,27 @@ impl ExecutionEnvironment {
         rng: &mut dyn RngCore,
         subnet_size: usize,
     ) -> Result<(), UserError> {
+        if let ThresholdArguments::Schnorr(ref schnorr) = args {
+            let alg = schnorr.key_id.algorithm;
+            match (alg, &schnorr.taproot_tree_root) {
+                (SchnorrAlgorithm::Bip340Secp256k1, Some(aux)) => {
+                    if aux.len() != 0 && aux.len() != 32 {
```

Comment:
Optionally you could add a test, similar to how the transport public key input sanitization is tested (see https://github.com/dfinity/ic/pull/4231).

Reply:
I added a test